sycl/doc/extensions/USM/USM.adoc

```diff
@@ -558,69 +680,56 @@ Return value:: none
 
 '''
 === Unified Shared Memory Information and Descriptors
-
-
-.Unified Shared Memory Pointer Query
+==== Pointer Queries
+===== get_pointer_type
 [source,cpp]
 ----
-std::tuple<usm::alloc, sycl::device*> get_pointer_info(const void* ptr);
+usm::alloc get_pointer_type(const void *ptr, const context &ctxt);
 ----
 Parameters::
   * `const void* ptr` - the pointer to query
-Return value:: Returns a `std::tuple` containing the type of the allocation and a pointer to the `device` against which it was allocated.  If this is a host allocation, the `device` pointer will be `nullptr`.
+  * `const sycl::context& ctxt` - the SYCL `context` to which the USM allocation belongs
+Return value:: Returns the USM allocation type for the allocation pointed to by `ptr`.  If `ctxt` is a host `context`, returns `usm::alloc::host`.
```

Comment:
What is the return value/behavior if `ptr` is `nullptr` or wasn't allocated by USM?

Does `ptr` have to be the beginning of an allocation (i.e. the value returned by a call to `malloc`) or can it be any address within a USM allocation?

Reply:
should be alloc::unknown.

Should work for any pointer in a USM allocated range.